cartographer/mapping/id.h

```diff
@@ -383,22 +383,26 @@ class MapById {
       return EndOfTrajectory(trajectory_id);
     }
 
-    size_t left = 0;
-    size_t right = trajectory.size() - 1;
+    auto left = trajectory.begin();
+    auto right = std::prev(trajectory.end());
     while (left != right) {
-      const int middle_index = (left + right) / 2;
-      auto middle = trajectory.begin();
-      std::advance(middle, middle_index);
```

Comment:
@mgladkova Thanks a lot for pointing out the bug. I changed the fix a bit, since this would change asymptotic behavior: this would take linear time for a `std::map`.